json/aerospike/6.4.0.json

```
@@ -0,0 +1,2881 @@
+              "properties": {
+                "type": {
+                  "type": "string",
+                  "description": "",
+                  "dynamic": false,
+                  "default": "flash",
+                  "enum": ["flash"]
+                },
+                "mounts": {
+                  "type": "array",
+                  "items": {
+                    "type": "string"
+                  },
+                  "description": "",
+                  "dynamic": false,
+                  "default": []
+                },
+                "mounts-high-water-pct": {
+                  "type": "integer",
+                  "default": 0,
+                  "minimum": 0,
+                  "maximum": 100,
+                  "description": "",
+                  "dynamic": true
+                },
+                "mounts-size-limit": {
+                  "type": "integer",
+                  "default": 0,
+                  "minimum": 4294967296,
+                  "maximum": 18446744073709551615,
+                  "description": "",
+                  "dynamic": true
+                }
+              }
+            }]
+          },
+          "sets": {
+            "type": "array",
+            "items": {
+              "type": "object",
+              "additionalProperties": false,
+              "properties": {
+                "name": {
+                  "type": "string",
+                  "default": " ",
+                  "description": "",
+                  "dynamic": false
+                },
+                "disable-eviction": {
+                  "type": "boolean",
+                  "default": false,
+                  "description": "",
+                  "dynamic": true
+                },
+                "enable-index": {
+                  "type": "boolean",
+                  "default": false,
+                  "description": "",
+                  "dynamic": true
+                },
+                "stop-writes-count": {
+                  "type": "integer",
+                  "default": 0,
+                  "minimum": 0,
+                  "maximum": 18446744073709551615,
+                  "description": "",
+                  "dynamic": true
+                },
+                "stop-writes-size": {
+                  "type": "integer",
+                  "default": 0,
+                  "minimum": 0,
+                  "maximum": 18446744073709551615,
+                  "description": "",
+                  "dynamic": true
+                }
+              }
+            }
+          },
+          "sindex-type": {
+            "oneOf": [{
+              "type": "object",
+              "additionalProperties": false,
+              "required": ["type"],
+              "properties": {
+                "type": {
+                  "type": "string",
+                  "description": "",
+                  "dynamic": false,
+                  "default": "shmem",
+                  "enum": ["shmem"]
```

```suggestion
                  "enum": ["shmem", "flash"]
```

Shouldn't this allow sindex-type "flash" because of https://aerospike.atlassian.net/browse/AER-6644

---

There is another section for the flash along with pmem. https://github.com/aerospike/schemas/blob/5e2af955c47c2e1e699981178cdabf00bb55b020/json/aerospike/6.4.0.json#L1954
Technically we can have an "enum": ["pmem", "flash"], but the server internally has different const/struct for both. So schema generator is also treating them differently. I'll see if making a common generator func to deal with pmem and Flash both make sense or not. May be in the next release.